sound/soc/sof/intel/hda.c

```diff
@@ -185,6 +185,10 @@ int hda_sdw_startup(struct snd_sof_dev *sdev)
 {
 	struct sof_intel_hda_dev *hdev;
 
+#if IS_ENABLED(CONFIG_SND_SOC_SDCA)
+	request_module("snd-soc-sdca");
+#endif
```

Comment:
to the commit message: maybe `hardware specific` instead of `vendor specific` drivers?
As for this specific change - it relies on the assumption, that if any matching hardware-specific drivers exist they would be loaded by now? This is run at post-firmware-load time